hooks/terraform_validate.sh

```diff
@@ -43,16 +43,88 @@ function per_dir_hook_unique_part {
   local -a -r args=("$@")
 
   local exit_code
-  local validate_output
+  #
+  # Get hook settings
+  #
+  local retry_once_with_cleanup=false
+
+  IFS=";" read -r -a configs <<< "${HOOK_CONFIG[*]}"
+
+  for c in "${configs[@]}"; do
+
+    IFS="=" read -r -a config <<< "$c"
+    key=${config[0]}
+    value=${config[1]}
+
+    case $key in
+      --retry-once-with-cleanup)
+        retry_once_with_cleanup=$value
+        ;;
+    esac
+  done
 
   common::terraform_init 'terraform validate' "$dir_path" || {
     exit_code=$?
     return $exit_code
   }
 
-  # pass the arguments to hook
-  validate_output=$(terraform validate "${args[@]}" 2>&1)
-  exit_code=$?
+  function do_validate {
+    validate_output=$(terraform validate "${args[@]}" 2>&1)
+    exit_code=$?
+    return $exit_code
+  }
+
+  function parse_validate {
+    # Requires jq
+    local exit_code
+    local validate_output
+    local valid
+    local summary
+
+    validate_output=$(terraform validate -json "${args[@]}" 2>&1)
+    exit_code=$?
+
+    valid="$(jq -rc '.valid' <<< "$validate_output")"
+
+    if [ "$valid" == "true" ]; then
+      return 0
+    fi
+
+    # Pretty-print error information
+    echo "$validate_output" | jq '.diagnostics[]'
```

Comment:
For consistency with surrounding code it may worth to use here-string here too:
```suggestion
    jq '.diagnostics[]' <<< "$validate_output"
```